dataprofiler/profilers/profile_builder.py

```diff
@@ -2088,11 +2088,13 @@ def _update_row_statistics(
 
         # If sample_ids provided, increment since that means only new data read
         if sample_ids is not None:
-            self.row_has_null_count += len(null_in_row_count)
-            self.row_is_null_count += len(null_rows)
+            if self.options.row_statistics.null_count.is_enabled:
+                self.row_has_null_count += len(null_in_row_count)
+                self.row_is_null_count += len(null_rows)
         else:
-            self.row_has_null_count = len(null_in_row_count)
-            self.row_is_null_count = len(null_rows)
+            if self.options.row_statistics.null_count.is_enabled:
+                self.row_has_null_count = len(null_in_row_count)
+                self.row_is_null_count = len(null_rows)
```

Comment:
I'd remove this as it is more "implementation code" versus just adding the option here

Reply:
just save this for the next PR as the "implementation PR" so its a bit smaller / chunking the code changes out for easier review